src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets

```diff
@@ -437,7 +437,7 @@ The .NET Foundation licenses this file to you under the MIT license.
       <CustomLinkerArg Include="-s ERROR_ON_UNDEFINED_SYMBOLS=0" />
       <CustomLinkerArg Condition="'$(NativeDebugSymbols)' != 'true'" Include="-O2 -flto" />
       <CustomLinkerArg Condition="'$(NativeDebugSymbols)' == 'true'" Include="-g3" />
-      <CustomLinkerArg Condition="$(NativeLib) == 'Shared'" Include="-Wl,--export,NativeAOT_StaticInitialization" />
+      <CustomLinkerArg Condition="$(NativeLib) != ''" Include="-Wl,--export,NativeAOT_StaticInitialization" />
```

Comment:
I am a bit ambivalent about this...

As https://github.com/dotnet/runtimelab/issues/2259#issuecomment-1520093017 mentions, we didn't originally enable it for all types of libraries because for static libraries the NAOT instructions say you should add it to the link manually.

Reply:
Ok, as we don't need it at all maybe we just remove it and leave it to the user to do ``-s EXPORTED_FUNCTIONS=_NativeAOT_StaticInitialization`

Reply:
I've left it as it was.

Reply:
Would be great to find the document on NAOT static libraries and add the linker parameter for exporting there.

Reply:
https://github.com/dotnet/runtimelab/blob/feature/NativeAOT-LLVM/samples/NativeLibrary/README.md  This one?

Reply:
I guess this section: https://github.com/dotnet/samples/blob/main/core/nativeaot/NativeLibrary/README.md#building-static-libraries, but we can't update it so it's fine as-is.